src/apis/Style.res

```diff
@@ -153,21 +140,146 @@ type justifyContent = [
+  textTransform?: textTransform,
+  writingDirection?: writingDirection,
+  // View styles https://reactnative.dev/docs/view-style-props
+  backfaceVisibility?: backfaceVisibility,
+  backgroundColor?: Color.t,
+  borderBottomColor?: Color.t,
+  borderBottomEndRadius?: float,
+  borderBottomLeftRadius?: float,
+  borderBottomRightRadius?: float,
+  borderBottomStartRadius?: float,
+  borderBottomWidth?: float,
+  borderColor?: Color.t,
+  borderEndColor?: Color.t,
+  borderEndWidth?: float,
+  borderLeftColor?: Color.t,
+  borderLeftWidth?: float,
+  borderRadius?: float,
+  borderRightColor?: Color.t,
+  borderRightWidth?: float,
+  borderStartColor?: Color.t,
+  borderStartWidth?: float,
+  borderStyle?: borderStyle,
+  borderTopColor?: Color.t,
+  borderTopEndRadius?: float,
+  borderTopLeftRadius?: float,
+  borderTopRightRadius?: float,
+  borderTopStartRadius?: float,
+  borderTopWidth?: float,
+  borderWidth?: float,
+  elevation?: float,
+  opacity?: float,
+  // Transform Props (https://reactnative.dev/docs/transforms#props)
+  transform?: array<transform>, // all other transform props are deprecated
+  // Shadow Props (https://reactnative.dev/docs/shadow-props)
+  shadowColor?: Color.t,
+  shadowOffset?: offset,
+  shadowOpacity?: float,
+  shadowRadius?: float,
+  // Layout Style Props (https://reactnative.dev/docs/layout-props)
+  alignContent?: alignContent,
+  alignItems?: alignItems,
+  alignSelf?: alignSelf,
+  aspectRatio?: float,
+  // border*Width are commented because already in view styles props (see explanation at the top)
+  // borderBottomWidth?: float,
+  // borderEndWidth?: float,
+  // borderLeftWidth?: float,
+  // borderRightWidth?: float,
+  // borderStartWidth?: float,
+  // borderTopWidth?: float,
+  // borderWidth?: float,
+  bottom?: size,
+  direction?: direction,
+  display?: display,
+  end?: size,
+  flex?: float,
+  flexBasis?: margin,
+  flexDirection?: flexDirection,
+  flexGrow?: float,
+  flexShrink?: float,
+  flexWrap?: flexWrap,
+  height?: size,
+  justifyContent?: justifyContent,
+  left?: size,
+  margin?: margin,
+  marginBottom?: margin,
+  marginEnd?: margin,
+  marginHorizontal?: margin,
+  marginLeft?: margin,
+  marginRight?: margin,
+  marginStart?: margin,
+  marginTop?: margin,
+  marginVertical?: margin,
+  maxHeight?: size,
+  maxWidth?: size,
+  minHeight?: size,
+  minWidth?: size,
+  overflow?: overflow,
+  padding?: size,
+  paddingBottom?: size,
+  paddingEnd?: size,
+  paddingHorizontal?: size,
+  paddingLeft?: size,
+  paddingRight?: size,
+  paddingStart?: size,
+  paddingTop?: size,
+  paddingVertical?: size,
+  position?: position,
+  right?: size,
+  start?: size,
+  top?: size,
+  width?: size,
+  zIndex?: int,
+}
+
+type styles = array<t>
+
+// Empty style. Cannot write it as {} in ReScript 10.0, this will be possible in 10.1.
+let empty = {color: ?None}
+
+let optional = style =>
+  switch style {
+  | Some(style) => style
+  | None => empty
+  }
```

Comment:
Can't we use identity to do option<t> => t ?

Reply:
No, because `t` is a record type now, but `None` is not a record.

Reply:
Yeah I get that. I was just thinking that RN styles accepts undefined, null & falsy values...

Reply:
Yeah but the user may do

```rescript
let x = Style.optional(None)
let color = x.color // 💥 runtime exception
```

Reply:
Yeah right, without abstract type, that's mandatory :)